src/cls/IPM/ResourceProcessor/WebApplication.cls

```diff
@@ -0,0 +1,263 @@
+     */
+    Set key = ""
+    While 1 {
+        Set value = ..ResourceReference.Attributes.GetNext(.key)
+        If (key = "") {
+            Quit
+        }
+        Set eval = ..%Evaluate(value)
+        Set ..SecurityProperties(key) = eval
+    }
+}
+
+Method ReplaceMatchRoles(matchRoles As %String, dbDir As %String) As %String
+{
+    New $NAMESPACE 
+    Set $NAMESPACE = "%SYS"
+    Set templates = $Listbuild("{$dbrole}", "${dbrole}")
+    For i=1:1:$Listlength(templates) {
+        Set template = $Listget(templates, i)
+        If matchRoles[template {
+            Set dbRole = "%DB_DEFAULT"
+            Set db = ##class(SYS.Database).%OpenId(dbDir)
+            If $Isobject(db) {
+                Set dbRole = db.ResourceName
+            }
+            Set matchRoles = $Replace(matchRoles, template, dbRole)
+        }
+    }
+    Return matchRoles
+}
+
+Method ReplaceNameSpace(expression As %String, namespace As %String) As %String
+{
+    Set templates = $Listbuild("{$ns}", "${ns}")
+    Set ptr = 0
+    While $LISTNEXT(templates, ptr, template) {
+        If expression[template {
+            Set expression = $REPLACE(expression, template, namespace)
+        }
+    }
+    Return expression
+}
+
+/// Called as phase <var>pPhase</var> is executed for the resource. If <var>pResourceHandled</var> is set to true,
+/// then the default behavior for that resource will be bypassed in the current phase.
+Method OnPhase(pPhase As %String, ByRef pParams, Output pResourceHandled As %Boolean = 0) As %Status
+{
+    Set sc = $$$OK
+    Try {
+        If (pPhase = "Validate") {
+            If (..Name = "") {
+                Return $$$ERROR($$$GeneralError,"Name is required for web applications")
+            }
+            Set keyList = ""
+            Set key = ""
+            While 1 {
+                Set value = ..ResourceReference.Attributes.GetNext(.key)
+                If (key = "") {
+                    Quit
+                }
+                Set keyList = keyList _ $ListBuild(key)
+            }
+            Set badKeyList = ..DoPropertiesExist(keyList)
+            If ($ListLength(badKeyList) > 0) {
+                Set error = "The following invalid attributes were found for resource "_..ResourceReference.Name_": "_$ListToString(badKeyList)
+                $$$ThrowStatus($$$ERROR($$$GeneralError,error))
+            }
+        }
+    } Catch (ex) {
+        Set sc = ex.AsStatus()
+    }
+    Return sc
+}
+
+Method OnBeforePhase(pPhase As %String, ByRef pParams) As %Status
+{
+    /*
+     * NOTE: Web app creation is done in OnBeforePhase so it happens before other
+     * resource processors such as FileCopy run.
+     */
+    Set sc = $$$OK
+    Try {
+        $$$ThrowOnError(##super(pPhase,.pParams))
+        
+        Set tVerbose = $Get(pParams("Verbose"))
+        #; Set tIsApplication = ..ResourceReference.Module.Lifecycle.%IsA(##class(%ZHSLIB.PackageManager.Developer.Lifecycle.Application).%ClassName(1))
```

Comment:
Can remove the tIsApplication check here and below along with the note about making it mirror safe. Mirror safety is the responsibility of the platform and IRIS will get there eventually with the new security database (coming in IRIS 2024.3 and will be mirror aware in 25.1 hopefully)

Reply:
Removed.